### PR TITLE
Unify Truffle Dashboard <Button /> component and provide new style properties - 4971

### PR DESCRIPTION
**Pull request by michaeljohnbennett**
Addresses #4971 

As per discussion with @MicaiahReid on adding in the refactored button code in a separate PR.

- Cleaned up imports to be all package relative.
- Also DashboardProvider.tsx renamed to be default packed export.
- Buttons cleaned up around site and disconnect button made a small red button.

We now have a Button element that will take styling of size/variants(filled/empty/etc)/colour.

### CODE REVIEW

**Comment by michaeljohnbennett** [comment]
going to revert the package relative imports as they fail on GitHub but work locally.

File: packages/dashboard/src/components/common/Button.tsx [hunk 1]
```diff
@@ -1,25 +1,108 @@
...
-interface Props {
-  disabled?: boolean;
-  onClick: MouseEventHandler<HTMLElement>;
-  text: string;
-}
+export type ButtonColor = "blue" | "red" | "brown" | "default";
+export type ButtonSize = "xs" | "sm" | "lg";
+export type ButtonVariant = "outlined" | "filled" | "empty";
+
+const DIMENSIONS = {
```
Comment (fainashalts) [inline packages/dashboard/src/components/common/Button.tsx:8]: Should these style-related aspects be in a separate file? I don't feel super strongly about it but that might be nicer.

Comment (gnidan) [reply packages/dashboard/src/components/common/Button.tsx:8]: Do they relate to anything outside this file? If they're only related to `<Button />` then they should probably live in this file

Comment (michaeljohnbennett) [reply packages/dashboard/src/components/common/Button.tsx:8]: its all button centric config. sorry I wasn't getting notices for this PR.

**Pull request merged**